vllm/config.py

```diff
@@ -1438,55 +1440,63 @@ def __post_init__(self):
 class ParallelConfig:
     """Configuration for the distributed execution."""
 
-    pipeline_parallel_size: int = 1  # Number of pipeline parallel groups.
-    tensor_parallel_size: int = 1  # Number of tensor parallel groups.
-    data_parallel_size: int = 1  # Number of data parallel groups.
-    data_parallel_rank: int = 0  # Rank of the data parallel group.
-    # Local rank of the data parallel group, defaults to global rank.
+    pipeline_parallel_size: int = 1
+    """Number of pipeline parallel groups."""
+    tensor_parallel_size: int = 1
+    """Number of tensor parallel groups."""
```

Comment:
Is it absolutely required for the comment to be after the statement? I would very much like the comments to be before the statement

Reply:
I asked the same thing about before/after on slack and Harry said:

> Yeah it does look weird initially... But it's correct https://peps.python.org/pep-0257/#what-is-a-docstring

> I think it's because the docstrings are being repurposed and normally they'd go under a function definition etc

> IMO it's worth it for the IDE integration alone

![image](https://github.com/user-attachments/assets/a35242ef-1862-41c4-aacc-329594c697d6)

Reply:
> Is it absolutely required for the comment to be after the statement?

Unfortunately, yes.

It's specified in [PEP 257](https://peps.python.org/pep-0257):
> 1. String literals occurring immediately after a simple assignment at the top level of a module, class, or __init__ method are called “attribute docstrings”.

This is the same as docstrings below function or class definitions.

Another advantage of doing it this way is that VS Code can see them:
![image](https://github.com/user-attachments/assets/9339d4ae-138c-42fd-bfd0-d9dc10e65fec)

---

> it would be nice if we had a unique trigger to specify that the comment is used for autogen docs

As it is now, only triple quoted `"""` comments will be grabbed as help strings for the argparser.

Reply:
Okay I understand. Those are nice arguments, thank you